File: web/src/api/docker/docker.js
import service from '@/utils/request'

export const getDockerServers = () => {
  return service({
    url: '/docker/servers',
    method: 'get'
  })
}

export const getDockerPs = (params) => {
  return service({
    url: '/docker/ps',
    method: 'get',
    params
  })
}

export const createContainer = (data) => {
  return service({ url: '/docker/createContainer', method: 'post', data })
}
export const createContainerByDockerfile = (data) => {
  return service({ url: '/docker/createContainerByDockerfile', method: 'post', data })
}
export const createContainerWithOptions = (data) => {
  return service({ url: '/docker/createContainerWithOptions', method: 'post', data })
}
export const startContainer = (params) => {
  return service({ url: '/docker/startContainer', method: 'post', params })
}
export const stopContainer = (params) => {
  return service({ url: '/docker/stopContainer', method: 'post', params })
}
export const removeContainer = (params) => {
  return service({ url: '/docker/removeContainer', method: 'delete', params })
}

export const getDockerLogs = (params) => {
  return service({ url: '/docker/logs', method: 'get', params })
}

export const getPreferredShell = (params) => {
  return service({ url: '/docker/preferredShell', method: 'get', params })
}

export const getDockerImages = (params) => {
  return service({ url: '/docker/images', method: 'get', params })
}

export const getServerAssetList = (params) => {
  return service({ url: '/SrvAst/getServerAssetList', method: 'get', params })
}

export const getServerAssetDataSource = () => {
  return service({ url: '/SrvAst/getServerAssetDataSource', method: 'get' })
}

File: server/router/docker/docker.go
package docker

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DockerRouter struct{}

// InitDockerRouter 初始化 Docker 服务管理 路由
func (r *DockerRouter) InitDockerRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	// 读接口（需鉴权）
	groupWithoutRecord := Router.Group("docker")
	{
		groupWithoutRecord.GET("servers", dockerApi.GetNormalServers)
		groupWithoutRecord.GET("ps", dockerApi.GetDockerPs)
		groupWithoutRecord.GET("preferredShell", dockerApi.PreferredShell)
		groupWithoutRecord.GET("logs", dockerApi.GetContainerLogs)
		groupWithoutRecord.GET("findDockerEndpoint", dockerEndpointApi.FindDockerEndpoint)
		groupWithoutRecord.GET("getDockerEndpointList", dockerEndpointApi.GetDockerEndpointList)
	}
	// 读接口（不鉴权）：WebSocket 终端
	groupPublic := PublicRouter.Group("docker")
	{
		groupPublic.GET("execWs", dockerApi.ContainerTerminalWs)
		groupPublic.GET("logsWs", dockerApi.ContainerLogsWs)
	}
	// 写接口
	group := Router.Group("docker").Use(middleware.OperationRecord())
	{
		group.POST("createDockerEndpoint", dockerEndpointApi.CreateDockerEndpoint)
		group.DELETE("deleteDockerEndpoint", dockerEndpointApi.DeleteDockerEndpoint)
		group.DELETE("deleteDockerEndpointByIds", dockerEndpointApi.DeleteDockerEndpointByIds)
		group.PUT("updateDockerEndpoint", dockerEndpointApi.UpdateDockerEndpoint)
		group.POST("createContainer", dockerApi.CreateContainer)
		group.POST("createContainerByDockerfile", dockerApi.BuildContainerByDockerfile)
		group.POST("createContainerWithOptions", dockerApi.CreateContainerWithOptions)
		group.POST("startContainer", dockerApi.StartContainer)
		group.POST("stopContainer", dockerApi.StopContainer)
		group.DELETE("removeContainer", dockerApi.RemoveContainer)
	}
}
